rank/services/search-templates.ts

```diff
@@ -93,34 +55,44 @@ export async function getLocalTemplates(): Promise<SearchTemplate[]> {
     works: works.default,
     images: images.default,
   }))
-
-  return searchTemplates.map((template) => ({
-    ...template,
-    id: `${template.namespace}/local/${template.index}`,
-    index: template.index,
-    env: 'local',
-    source: { query: queries[template.namespace] },
-  }))
+  return queries
 }
 
-/**
- * This service merges remote and local search templates.
- *
- * A local search template is available when there is an existing index
- * with a corresponding query in `./data/queries/{index}.json`.
- */
-type Props = {
-  prod?: true
-  local?: true
-  test?: string[]
+export async function getQueries() {
+  const queries = {
+    local: await getLocalQueries(),
+    remote: await getRemoteQueries(),
+  }
+  return queries
 }
-export async function getTemplates({ prod, local, test }: Props = {}): Promise<
-  SearchTemplate[]
-> {
-  const remoteTemplates = prod ? await getRemoteTemplates('prod') : []
-  const localTemplates = local ? await getLocalTemplates() : []
-  const templates = remoteTemplates.concat(localTemplates)
+
+export async function getTemplates(
+  ids?: SearchTemplateString[]
+): Promise<SearchTemplate[]> {
+  if (!ids) {
+    const indices = await listIndices()
+    ids = envs.flatMap((env) =>
+      indices.map((index) => `${env}/${index}` as SearchTemplateString)
+    )
+  }
+  const queries = await getQueries()
+  const templates = ids
+    .map((id) => new SearchTemplate(id))
+    .map((template) => {
+      template.query = queries[template.env][template.namespace]
```

Comment:
Mutating objects like this if often a sign we might want to do it differently.

Reply:
This seemed like the neatest way around the async-class-creation problem we talked about previously. Happy to keep pulling that thread if you're not comfortable with this though!